src/librustc_data_structures/indexed_vec.rs

```diff
@@ -149,7 +149,7 @@ macro_rules! newtype_index {
 
             #[inline]
             $v const unsafe fn from_u32_unchecked(value: u32) -> Self {
-                unsafe { $type { private: value } }
+                $type { private: value }
```

Comment:
it's not needed because the function itself is `unsafe`, so it's allowed to have unsafe operations in the body

Reply:
Fair enough. The obvious follow-up question is "why didn't the compiler complain about this before?"